src/coreComponents/finiteVolume/FluxApproximationBase.cpp

```diff
@@ -63,22 +63,17 @@ void FluxApproximationBase::initializePreSubGroups()
 {
   DomainPartition & domain = this->getGroupByPath< DomainPartition >( "/Problem/domain" );
 
-  Group & meshBodies = domain.getMeshBodies();
-
-  meshBodies.forSubGroups< MeshBody >( [&]( MeshBody & meshBody )
+  for( MeshLevel * mesh: domain.getMeshBodies() | flattenMeshLevels() )
```

Comment:
is there a drawback to the nested looping? I don't know that I understand the motivation to flatten?

Reply:
Not really, it's just that eventually we only loop on the mesh levels, so having one loop is fine too _imho_.

Flattening reduces the "_depth_" of the function which can always be nice: sometimes we do have deep functions and it's complicated to know where we are. No big deal but it can be nice.

In the current case of `FluxApproximationBase`, this changes a double loop into a simple one (same real complexity of course). Not a critical improvement.

We could also rename into
```c++
  for( MeshLevel * mesh: domain.getMeshBodies() | getMeshLevels() ) // or GetMeshLevels()
```
for more obvious reading (in the end, this flattening is a reading concern).

_One point which I find of interest though_: the `meshBody` instance which was only used to retrieve the mesh levels is now invisible. I think it's interesting.

But let's be honest, my main idea was to mimic the `flatMap` classical pattern which
- perfectly matches here,
- can be really fluid to read (personal opinion).